### PR TITLE
luaPackages.awesome-wm-widgets: init at a83ba6ff7

### PR DESCRIPTION
**Pull request by rski**
###### Motivation for this change

awesome-wm-widgets is a fairly widely used set of widgets for Awesome WM.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/lua-packages.nix [hunk 1]
```diff
@@ -132,6 +138,43 @@ with self; {
...
+
+    buildInputs = [ lua ];
+
+    postPatch = ''
+      rm *.png
+      find -iname "screenshot*" | xargs rm -rf
+      find -iname "example*" | xargs rm
+      find -iname "README.md" | xargs rm
+      find -iname "*gif" | xargs rm
+      find -iname "*py" | xargs rm
```
Comment (SuperSandro2000) [inline pkgs/top-level/lua-packages.nix:160]: This can be combined, right?

File: pkgs/top-level/lua-packages.nix [hunk 2]
```diff
@@ -132,6 +138,43 @@ with self; {
...
+      rm -rf experiments mpris-widget # mpris is broken upstream
```
Comment (SuperSandro2000) [inline pkgs/top-level/lua-packages.nix:161]:
```suggestion
      rm -r experiments mpris-widget # mpris is broken upstream
```

File: pkgs/top-level/lua-packages.nix [hunk 3]
```diff
@@ -132,6 +138,43 @@ with self; {
     };
   };
 
+  awesome-wm-widgets = toLuaModule(stdenv.mkDerivation rec {
+    pname = "awesome-wm-widgets";
+    version = "a83ba6ff7";
```
Comment (SuperSandro2000) [inline pkgs/top-level/lua-packages.nix:143]: Please never use short shas anywhere longterm. Either use the full sha or a tag. Also version should be in unstable-XXXX-XX-XX format.

File: pkgs/top-level/lua-packages.nix [hunk 4]
```diff
@@ -46,6 +47,11 @@ let
       };
     });
 
+  widgetInstallPhase = pname: ''
+     mkdir -p $out/lib/lua/${lua.luaversion}/
+      cp -r . $out/lib/lua/${lua.luaversion}/${pname}/
+      printf "package.path = '$out/lib/lua/${lua.luaversion}/?/init.lua;' ..  package.path\nreturn require((...) .. '.init')\n" > $out/lib/lua/${lua.luaversion}/${pname}.lua
+    '';
```
Comment (SuperSandro2000) [inline pkgs/top-level/lua-packages.nix:54]:
```suggestion
     mkdir -p $out/lib/lua/${lua.luaversion}/
     cp -r . $out/lib/lua/${lua.luaversion}/${pname}/
     printf "package.path = '$out/lib/lua/${lua.luaversion}/?/init.lua;' ..  package.path\nreturn require((...) .. '.init')\n" > $out/lib/lua/${lua.luaversion}/${pname}.lua
   '';
```

File: pkgs/top-level/lua-packages.nix (hunk 3 above)
Comment (rski) [reply pkgs/top-level/lua-packages.nix]: ah ok, I was wondering about what to use for a version. Should be done now.

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 135272` at 855c6571 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/8c2ef5271e5a280fffb496d725d6b0ed">lua51Packages.awesome-wm-widgets</a></li>
    <li><a href="https://gist.github.com/414cc05bd5582f33aa7c9645b4e2401d">luaPackages.awesome-wm-widgets</a> (lua52Packages.awesome-wm-widgets)</li>
    <li><a href="https://gist.github.com/5cf42a6c7bb6bb66709c9bba63f0815d">lua53Packages.awesome-wm-widgets</a></li>
    <li><a href="https://gist.github.com/225453bb162dca02508d18c3023fa550">luajitPackages.awesome-wm-widgets</a></li>
  </ul>
</details>
<details>
  <summary>4 packages built successfully:</summary>
  <ul>
    <li>lua51Packages.vicious</li>
    <li>luaPackages.vicious (lua52Packages.vicious)</li>
    <li>lua53Packages.vicious</li>
    <li>luajitPackages.vicious</li>
  </ul>
</details>
<details>
  <summary>13 suggestions:</summary>
  <ul>
    <li>warning: unused-argument

Unused argument: `gnulib`.
Near pkgs/top-level/lua-packages.nix:9:20:
```
  |
9 | , pcre, oniguruma, gnulib, tre, glibc, sqlite, openssl, expat
  |                    ^
```
</li>
    <li>warning: unused-argument

Unused argument: `expat`.
Near pkgs/top-level/lua-packages.nix:9:57:
```
  |
9 | , pcre, oniguruma, gnulib, tre, glibc, sqlite, openssl, expat
  |                                                         ^
```
</li>
    <li>warning: unused-argument

Unused argument: `cyrus_sasl`.
Near pkgs/top-level/lua-packages.nix:11:15:
```
   |
11 | , postgresql, cyrus_sasl
   |               ^
```
</li>
    <li>warning: unused-argument

Unused argument: `pcre`.
Near pkgs/top-level/lua-packages.nix:9:3:
```
  |
9 | , pcre, oniguruma, gnulib, tre, glibc, sqlite, openssl, expat
  |   ^
```
</li>
    <li>warning: unused-argument

Unused argument: `oniguruma`.
Near pkgs/top-level/lua-packages.nix:9:9:
```
  |
9 | , pcre, oniguruma, gnulib, tre, glibc, sqlite, openssl, expat
  |         ^
```
</li>
    <li>warning: unused-argument

Unused argument: `unzip`.
Near pkgs/top-level/lua-packages.nix:8:26:
```
  |
8 | { fetchurl, stdenv, lua, unzip, pkg-config
  |                          ^
```
</li>
    <li>warning: unused-argument

Unused argument: `openssl`.
Near pkgs/top-level/lua-packages.nix:9:48:
```
  |
9 | , pcre, oniguruma, gnulib, tre, glibc, sqlite, openssl, expat
  |                                                ^
```
</li>
    <li>warning: unused-argument

Unused argument: `glibc`.
Near pkgs/top-level/lua-packages.nix:9:33:
```
  |
9 | , pcre, oniguruma, gnulib, tre, glibc, sqlite, openssl, expat
  |                                 ^
```
</li>
    <li>warning: unused-argument

Unused argument: `tre`.
Near pkgs/top-level/lua-packages.nix:9:28:
```
  |
9 | , pcre, oniguruma, gnulib, tre, glibc, sqlite, openssl, expat
  |                            ^
```
</li>
    <li>warning: unused-argument

Unused argument: `postgresql`.
Near pkgs/top-level/lua-packages.nix:11:3:
```
   |
11 | , postgresql, cyrus_sasl
   |   ^
```
</li>
    <li>warning: unused-argument

Unused argument: `autoreconfHook`.
Near pkgs/top-level/lua-packages.nix:10:3:
```
   |
10 | , autoreconfHook, gnum4
   |   ^
```
</li>
    <li>warning: unused-argument

Unused argument: `gnum4`.
Near pkgs/top-level/lua-packages.nix:10:19:
```
   |
10 | , autoreconfHook, gnum4
   |                   ^
```
</li>
    <li>warning: unused-argument

Unused argument: `sqlite`.
Near pkgs/top-level/lua-packages.nix:9:40:
```
  |
9 | , pcre, oniguruma, gnulib, tre, glibc, sqlite, openssl, expat
  |                                        ^
```
</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

**Comment by rski** [comment]
mhhh fixed point derivations, probably my biggest complaint about nix :<

**Review by zt610588 — APPROVED** [review]

**Comment by rski** [comment]
I'm declaring pr bankruptcy for now, if anyone else wants these, here is [a link to my dotfiles  ](https://gitlab.com/rski/dotfiles/-/blob/master/nixpkgs/desktop.nix#L275)

**Pull request closed without merging**